### ISSUE TITLE
BTT EXP-MOT with MKS Robin E3D

### ISSUE THREAD

**Issue by tanjunior** [bigtreetech/BTT-Expansion-module#28]
I wanted to add 1 additional stepper driver to my MKS ROBIN E3D so that I can do Z_TILT.

E3D UART mode
MOT-EXP UART mode
E3D EXP1 <-> EXP-MOT EXP1
E3D EXP2 <-> EXP-MOT EXP2

Following the [BTT EXP-MOT user guide](https://github.com/bigtreetech/BTT-Expansion-module/blob/master/BTT%20EXP-MOT/BTT%20EXP-MOT%20V1.0User%20Guide.pdf) and cross referencing with the [E3D pins](https://github.com/makerbase-mks/MKS-Robin-E3-E3D/blob/master/hardware/MKS%20Robin%20E3D%20V1.1_003/MKS%20Robin%20E3D%20V1.1_003%20PIN.pdf)

BTT-MOT EXP2 M1STP = E3D EXP2 PB15 (klipper alias EXP2_6)
BTT-MOT EXP2 M1DIR = E3D EXP2 PB0 (klipper alias EXP2_5)
BTT-MOT EXP2 M1EN = E3D EXP2 PC10 (klipper alias EXP2_7)
BTT-MOT EXP1 M1RX = E3D EXP1 PA7 (klipper alias EXP1_6)

```
[stepper_z1]
step_pin: EXP2_6
dir_pin: !EXP2_5
enable_pin: !EXP2_7
microsteps: 16
rotation_distance: 8
endstop_pin: ^PC6
endstop_pin: probe:z_virtual_endstop

[tmc2209 stepper_z1]
uart_pin: EXP1_6
run_current: 0.8
stealthchop_threshold: 999999

[board_pins]
aliases:
    # EXP1 header
    EXP1_1=PC1, EXP1_3=PA4, EXP1_5=PA6, EXP1_7=PC4, EXP1_9=<GND>,
    EXP1_2=PC3, EXP1_4=PA5, EXP1_6=PA7, EXP1_8=PC5, EXP1_10=<5V>,
    # EXP2 header
    EXP2_1=PB14, EXP2_3=PB11, EXP2_5=PB0,  EXP2_7=PC10,  EXP2_9=<GND>,
    EXP2_2=PB13, EXP2_4=PA15, EXP2_6=PB15, EXP2_8=<RST>, EXP2_10=<NC>,
    # CR_EXP3 header - additional header dedicated to Creality Ender 3/5 stock screens
    CR_EXP3_10=<5V>, CR_EXP3_8=PA4, CR_EXP3_6=PA6, CR_EXP3_4=<RST>, CR_EXP3_2=PC3,
    CR_EXP3_9=<GND>, CR_EXP3_7=PA5, CR_EXP3_5=PB0, CR_EXP3_3=PB11,  CR_EXP3_1=PC1
```

`FORCE_MOVE STEPPER=stepper_z1 DISTANCE=10 VELOCITY=5`
![image](https://github.com/user-attachments/assets/5bc27f0e-6a57-4aaa-b78f-c1e7d9800d1d)

**Comment by VORONENTHUSIAST** [comment]
Can you show how your plugs are inserted into both boards, please?

Looking at the pinout you've shown against the pinout for the BTT EXP-MOT, they are reversed. 
Pin 1 is is bottom left for the BTT board when looking at it directly, Pin 1 is top right on your MKS board - this is the same as my SKIPR Board. 

Other than that all I can think is maybe not use aliases? It seems much more logical to just use the pin names from your MKS E3D and hash the EXP number next to it.

![MOT-EXP M1 Port](https://github.com/user-attachments/assets/dd21a39a-0a4a-437a-9d5c-676e30ab565d)
![EXP-MOT M1 Port](https://github.com/user-attachments/assets/4df179af-7016-4d9d-b04b-d732e7307f29)

That's extruder code taken from the working file of one of my Voron 2.4s.

TLDR: BTT and MKS EXP ports are reversed, the cables of one side (I chose mainboard) must have their tabs cut off and be rotated 180 degrees. If you already did this, just try not using aliases and see what happens?

**Comment by tanjunior** [comment]
> Can you show how your plugs are inserted into both boards, please?

![image](https://github.com/user-attachments/assets/b6ee90cd-e669-48eb-81df-f8d192983e99)

> the cables of one side (I chose mainboard) must have their tabs cut off and be rotated 180 degrees.

the port on my MKS side does not have a housing surrounding it, allows me to connect in either orientation.

>try not using aliases and see what happens?

I will try and give you an update

**Comment by tanjunior** [comment]
@VORONENTHUSIAST Still getting the same error without using alias.

**Comment by VORONENTHUSIAST** [comment]
> @VORONENTHUSIAST Still getting the same error without using alias.

I'm all out of ideas then I'm afraid.
Mine is up and running using the EXP ports I can see in your shared config.

Did you ever plug it in the wrong way around? I'm just wondering if you've popped the board perhaps. 
All wires are plugged in for both EXP ports?

**Comment by tanjunior** [comment]
> Did you ever plug it in the wrong way around? I'm just wondering if you've popped the board perhaps.

What do you meant by popped?

> All wires are plugged in for both EXP ports?

yes both are plugged in!

Also wondering if defining an extra MCU is needed

**Comment by VORONENTHUSIAST** [comment]
By popped, I mean damaged.

No defining of an MCU is needed - the BTT EXP-MOT doesn't have a control chip like your main board

**Comment by tanjunior** [comment]
> By popped, I mean damaged.

I notice that the Z2 stepper jerks when I power on my printer. Other than that no smell and onboard LEDS are on.

**Comment by VORONENTHUSIAST** [comment]
Are you using all 3 ports on the board, if so are the other 2 working?

**Comment by tanjunior** [comment]
> Are you using all 3 ports on the board, if so are the other 2 working?

I only need 1 I have only tested with M1 so far. Now that you mentioned it I should had tested the other 2

**Comment by tanjunior** [comment]
```
[stepper_z1] #m1
step_pin: PB15 #EXP2_6
dir_pin: !PB0 #!EXP2_5
enable_pin: !PC10 #!EXP2_7
microsteps: 16
rotation_distance: 8
endstop_pin: ^PC6
endstop_pin: probe:z_virtual_endstop

[stepper_z2] #m2
step_pin: EXP2_3
dir_pin: !EXP2_4
enable_pin: !EXP1_8
microsteps: 16
rotation_distance: 8
endstop_pin: ^PC6
endstop_pin: probe:z_virtual_endstop

[stepper_z3] #m3
step_pin: EXP2_1
dir_pin: !EXP2_2
enable_pin: !EXP1_7
microsteps: 16
rotation_distance: 8
endstop_pin: ^PC6
endstop_pin: probe:z_virtual_endstop

[tmc2209 stepper_z1] #m1
uart_pin: PA7 #EXP1_6
# diag_pin: PA6 #EXP1_5
run_current: 0.8
stealthchop_threshold: 999999

[tmc2209 stepper_z2] #m2
uart_pin: EXP1_4
# diag_pin: EXP1_3
run_current: 0.8
stealthchop_threshold: 999999

[tmc2209 stepper_z3] #m3
uart_pin: EXP1_2
# diag_pin: EXP1_1
run_current: 0.8
stealthchop_threshold: 999999
```

I have tried all 3 and it's not working.

**Comment by VORONENTHUSIAST** [comment]
No clue then unfortunately, as I said, mine is working

**Comment by kaswets** [comment]
I have a octopus Pro board and a exp-mot.

can I run Uart mode on the Exp-mot?  and what pins need I to select in klipper printer.cfg?

Please help!
